docs/rfcs/1013-parsec-account.md

```diff
@@ -0,0 +1,507 @@
+
+```yaml
+{
+  "label": "AuthMethod",
+  "type": "auth_method",
+  "other_fields": {
+    "algorithm": "KeyAlgorithm",
+    "hmac_key": "HMACKey",
+    "account_id": "AccountID",
+    "vault_id": "VaultID",
+    "vault_encrypted_key": "EncryptedSymKey",
+    "created_on": "Timestamp",
+    "created_by": "Creator"
+  }
+}
+```
+- The algorithm field contains the full configuration needed to do the derivations operations. Typically, for the password it should be `{"name": "Argon2iBlake2", "argon2i_opslimit": 65536, "argon2i_memlimit_kb": 3, "argon2i_parallelism": 1}`
+- There can be only a single password authentication method per vault.
+- Other authentication methods (such as FIDO2) may be added in the future.
+
+#### Protected Device
+
+[`ProtectedDevice`]: #protected-device
+
+```yaml
+{
+  "label": "ProtectedDevice",
+  "type": "protected_device",
+  "other_fields": {
+    "organization_id": "OrganizationID",
+    "device_id": "DeviceID",
+    "vault_id": "ID",
+    "encrypted_device": "EncryptedDevice"
+  }
+}
+```
+- Device are protected within a vault
+- Any given device can be protected in multiple vaults (this is typically the case when the devices have been recovered from an older vault).
+
+### API
+
+#### Account creation
+
+To create an account, the client starts by requesting an email validation token.
+
+```yaml
+{
+  "cmd": "account_send_email_validation_token",
+  "req": {
+    "email": "EmailStr"
+  },
+  "reps": [
+    {
+      "status": "ok"
+    },
+    {
+      "status": "invalid_email"
+    }
+  ]
+}
+```
+
+On `ok`, the server would have sent a mail with a unique token used for next the request used to register the authentication method.
+
+> The token will be valid for a duration that will be defined at the implementation (likely will be configurable)
+
+If the email is already registered in the service, a `ok` response will still be sent (without sending an email)
+to avoid creating an oracle about emails registered in the service.
+
+```yml
+{
+  "cmd": "account_password_create",
+  "req": {
+    "email_validation_token": "EmailValidationToken",
+    "account_label": "String",
+    "auth_method_key_params": "KeyAlgorithm",
+    "auth_method_hmac_key": "HMACKey",
+    "encrypted_vault_key": "EncryptedSymKey"
+  },
+  "reps": [
+    {
+      "status": "ok",
+    },
+    {
+      "status": "invalid_email_validation_token"
+    }
+  ]
+}
+```
+
+Upon receiving the request, the server will create the [`Account`], [`AuthMethod`], and [`Vault`].
+
+The IP address and User-agent of the creator are recorded (see Data Model).
+
+#### Uploading a new device
+
+To upload a new device the user need to retrieve the current vault information
+
+```yaml
+{
+  "cmd": "vault_get",
+  "req": {},
+  "reps": {
+    "ok": {
+      "vault_id": "VaultID",
+      "encrypted_key": "EncryptedSymKey",
+      "created_on": "Timestamp",
+      "created_by": "Creator",
+      # True if using the active vault (latest vault).
```

Comment:
I'm not sure how we could get not the latest vault here

Reply:
Currently it will always be true since it will always return the latest known vault. 
I'm anticipating that we add a `id: Option<VaultID>` in the request field later on